contracts/SpokePool.sol

```diff
@@ -1000,13 +1008,19 @@ abstract contract SpokePool is
         // This allows the caller to add in frontrunning protection for quote validity.
         require(fillCounter[relayData.destinationToken] <= relayExecution.maxCount, "Above max count");
 
-        // Stores the equivalent amount to be sent by the relayer before fees have been taken out.
+        // If the max tokens to send is 0, then the fillAmountPreFees will be 0 so we exit early.
+        // This also means that if the ending filled amount is 0, then we won't enter the
+        // function _updateCountFromFill. Therefore, _updateCountFromFill doesn't explicilty check
+        // that the ending fill amount is 0 (i.e. that the fillAmountPreFees is 0).
+        // If this if statement were to be removed, then _updateCountFromFill would need to be updated to
+        // check that the ending fill amount is 0.
         if (relayExecution.maxTokensToSend == 0) return 0;
 
         // Derive the amount of the relay filled if the caller wants to send exactly maxTokensToSend tokens to
         // the recipient. For example, if the user wants to send 10 tokens to the recipient, the full relay amount
         // is 100, and the fee %'s total 5%, then this computation would return ~10.5, meaning that to fill 10.5/100
         // of the full relay size, the caller would need to send 10 tokens to the user.
+        // This is equivalent to the amount to be sent by the relayer before fees have been taken out.
         fillAmountPreFees = _computeAmountPreFees(
```

Comment:
@mrice32 Is it possible that `_computeAmountPreFees` could set `fillAmountPreFees` to 0if the `maxTokensToSend` is really small or the `(relayData.realizedLpFeePct + relayExecution.updatedRelayerFeePct)` is really large?

If so, then this PR would no longer catch the case where  `relayFills[relayExecution.relayHash] = 0 = relayFills[relayExecution.relayHash] + fillAmountPreFees`.

I think it should be fine but wanted to get eyes on this

Reply:
Great call-out. I'll need to think on this.